File: src/main_handler.erl
-module(main_handler).

-export([init/2]).

init(Req, Opts) ->
	Body = jsx:encode(#{
		<<"messages">> => [
			hello, world
		],
		<<"timestamp">> => timestamp()
	}),
	Req1 = cowboy_req:reply(200, [
		{<<"content-type">>, <<"application/json">>},
		{<<"connection">>, <<"close">>}
	], << Body/binary, <<"\n">>/binary >>, Req),
	{ok, Req1, Opts}.

%% ===================================================================
%% Internal
%% ===================================================================

timestamp() ->
	{M, S, _} = os:timestamp(),  
	M * 1000000 + S.

File: lib/main_handler.ex
defmodule MainHandler do

	def init(request, state) do
		{:ok, body} = JSX.encode(%{
			"messages" => [:hello, :world]
		})

		reply = :cowboy_req.reply(200, [
			{"content-type", "application/json"}
		], body, request)

		{:ok, reply, state}
	end

end
